rational2crafted.el

```diff
@@ -0,0 +1,120 @@
+;;; rational2crafted.el --- Convert rational-emacs to crafted-emacs  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2022
+;; SPDX-License-Identifier: MIT
+
+;; Author: System Crafters Community
+
+;;; Commentary:
+
+;; Convert a Rational Emacs configuration to a Crafted Emacs
+;; configuration. This involves the following steps:
+;;
+;; 1. Find the `rational-config-path' and copy it to the new crafted
+;; path.
+;;
+;; 2. Set the `crafted-config-path' from the `rational-config-path'
+;;
+;; 3. Replace `rational' with `crafted' in the users `config.el' and
+;; `early-config.el', assuming the latter exists.
+;;
+;; 4. Check if there are any modules in the users `custom-modules'
+;; folder, and replace any `rational' with `crafted' including
+;; renaming the modules if necessary.
+;;
+;; Usage:
+;;
+;; emacs -Q --batch -l rational2crafted.el
+
+;;; Code:
+
+(require 'xdg)
+(require 'files)
+(require 'seq)
+
+;; Find the rational-config-path.  This will be used as the basis for
+;; the crafted-config-path.  We find the the `rational-config-path'
+;; the same way the `early-init.el' file does.
+(defvar rational-config-path
+  (cond
+   ((featurep 'chemacs)
+    (if (getenv  "RATIONAL_EMACS_HOME")
+        (expand-file-name (getenv "RATIONAL_EMACS_HOME"))
+      (expand-file-name "rational-emacs" user-emacs-directory)))
+   ((getenv "RATIONAL_EMACS_HOME") (expand-file-name (getenv "RATIONAL_EMACS_HOME")))
+   ((or (file-exists-p (xdg-config-home))
+        (file-exists-p (expand-file-name "rational-emacs" (xdg-config-home))))
+    (expand-file-name "rational-emacs" (xdg-config-home)))
+   ((getenv "HOME") (expand-file-name ".rational-emacs" (getenv "HOME"))))
+  "The user's rational-emacs configuration path.")
+
+(print (concat "rational config path: " rational-config-path))
+
+;; Set the `crafted-config-path' from the `rational-config-path'
+(defvar crafted-config-path
+  (string-replace "rational" "crafted" rational-config-path)
+  "The user's crafted-emacs config path.")
+
+(print (concat "crafted config path: " crafted-config-path))
+
+;; Copy the rational-emacs folder to crafted-emacs.  This,
+;; effectively, makes a backup of the rational-config-path. From here,
+;; we will only work with in the crafted-config-path folder.
+(copy-directory rational-config-path crafted-config-path)
+
+(defvar crafted-config-file
+  (expand-file-name "config.el" crafted-config-path)
+  "The user's crafted-emacs config file, config.el")
+
+(defvar crafted-early-config-file
+  (expand-file-name "early-config.el" crafted-config-path)
+  "The user's crafted-emacs early init file, early-config.el")
+
+(print (concat "crafted-config-file: " crafted-config-file))
+(print (concat "crafted-early-config-file: " crafted-early-config-file))
+
+(defun perform-replacement ()
+  (goto-char (point-min))
+  (while (re-search-forward "rational" nil t)
```

Comment:
I wonder if we should search for `rational-` since all variables, functions, and libraries use a hyphen after?  I know this might not catch other usages of `rational*` but it feels safer to not catch every instance of the word.